account_move_stripe_import/models/account_move.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2020 Akretion (https://www.akretion.com).
+# @author Pierrick Brun <pierrick.brun@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+from odoo.addons.account_move_base_import.models.account_move import (
+    ErrorTooManyPartner,
+)
+
+
+class AccountMoveCompletionRule(models.Model):
+    """Add a rule based on transaction ID"""
+
+    _inherit = "account.move.completion.rule"
+
+    function_to_call = fields.Selection(
+        selection_add=[
+            (
+                "get_from_transaction_ref_and_so",
+                "Match Sales Order using transaction Reference",
+            ),
+        ]
+    )
+
+    def get_from_transaction_ref_and_so(self, line):
+        """
+        Same as get_from_transaction_id_and_so but uses the reference field
+        of the transaction instead of its ID.
+        """
+        res = {}
+        so_obj = self.env["sale.order"]
+        reference = line.name
+        if line.name.startswith("REFUND"):
+            # Substring between
+            reference = line.name[len("REFUND FOR CHARGE(")+1:-len(")")]
+        sales = so_obj.search([("transaction_ids.reference", "=", reference)])
+        partners = sales.mapped("partner_id")
+        if len(partners) > 1:
+            raise ErrorTooManyPartner(
+                _('Line named "%s" was matched by more than ' "one partner.")
+                % line.name
+            )
+        if len(partners) == 1:
+            res["partner_id"] = partners.id
+            res["account_id"] = partners.property_account_receivable_id.id
```

Comment:
I'm not sure we should change account_id to put the receivable account.

The invoice is accounted on receivable account and the payment (account.payment) linked to the invoice is reconciled with the invoice. Receivable account balance is then 0€.

I think we'd better let the account as it is when importing the statement (the account configured on account.journal.receivable_account_id).

Reply:
I'm not sure to understand were to find the data from where we are here.

If you want you can do a PR against mine and I'll merge it

Reply:
Hi @PierrickBrun 

> I'm not sure to understand were to find the data from where we are here.
> 
> If you want you can do a PR against mine and I'll merge it

Is https://github.com/akretion/account-reconcile/pull/2 all good ?